reflex/app.py

```diff
@@ -659,81 +654,128 @@ def compile(self):
             # If a theme component was provided, wrap the app with it
             app_wrappers[(20, "Theme")] = self.theme
 
-        with progress:
-            for route, component in self.pages.items():
-                # TODO: this progress does not reflect actual threaded task completion
+        with progress, concurrent.futures.ThreadPoolExecutor() as thread_pool:
+            fixed_pages = 7
+            compile_total = sum(
+                [
+                    len(self.pages),  # Pre-processing
+                    len(self.pages),  # Memoize stateful components
+                    len(self.pages) + fixed_pages,  # Compile pages
+                    len(self.pages) + fixed_pages,  # Write pages
```

Comment:
I don't think we need to separate out all the writes, they happen pretty instantly. We should remove them or combine them to one step